premium: add database audit logging (#22044)
pingcap/docs: tidb-cloud/premium/tidb-cloud-auditing-premium.md

---
title: "{{{ .premium }}} Database Audit Logging"
summary: Learn how to audit an instance in {{{ .premium }}}.
---

# {{{ .premium }}} Database Audit Logging

> TiDB Cloud provides an audit logging feature that records user access activities, such as any executed SQL statements.

Reviewer: reason: to distinguish it from console audit logging

1. In the **Bucket URI** field, enter the URI of your S3 bucket where the audit log files are to be written.
2. In the **Bucket Region** drop-down list, select the AWS region where the bucket is located.
3. In the **Role ARN** field, fill in the Role ARN value that you copied in [Step 2. Configure Amazon S3 access](#step-2-configure-amazon-s3-access).
4. Click **Test Connection and Next** to verify whether TiDB Cloud can access and write to the bucket.

    If it is successful, **The connection is successful** is displayed. Otherwise, check your access configuration.

5. Click **Enable** to enable audit logging for the instance.

    TiDB Cloud is ready to write audit logs for the specified instance to your Amazon S3 bucket.

> **Note:**
>
> - After enabling audit logging, if you make any new changes to the bucket URI, location, or ARN, you must click **Test Connection** again to verify that TiDB Cloud can connect to the bucket. Then, click **Enable** to apply the changes.
> - To remove TiDB Cloud's access to your Amazon S3, simply delete the trust policy granted to this instance in the AWS Management Console.

> ### Enable audit logging for TiDB on Alibaba Cloud

Reviewer: Not sure if we need to wrap this section with CustomContent, just in case we forget to do so when Premium goes into public preview in the future.

```
<CustomContent language="en,zh">

</CustomContent>
```

To enable database audit logging for TiDB cloud on Alibaba Cloud, take the following steps:

#### Step 1. Create an OSS bucket

Create an Object Storage Service (OSS) bucket in your organization-owned Alibaba Cloud account as the destination to which TiDB Cloud writes audit logs.

> For more information, see [Creating storage buckets](https://www.alibabacloud.com/help/en/oss/user-guide/create-a-bucket-4) in the Alibaba Cloud Storage documentation.

Reviewer: ```suggestion
For more information, see [Create a bucket](https://www.alibabacloud.com/help/en/oss/user-guide/create-a-bucket-4) in the Alibaba Cloud Storage documentation.
```

    3. On the **DB Audit Logging** page, click **Enable** in the upper-right corner.
    4. In the **Database Audit Log Storage Configuration** dialog, locate the **Alibaba Cloud RAM Policy Settings** section, and record **TiDB Cloud Account ID** and **TiDB Cloud External ID** for later use.

2. In the Alibaba Cloud console, go to **RAM** > **Permissions** > **Policies**, and then check whether a policy already exists with the `oss:PutObject` write-only permission for your audit log bucket.

    - If yes, record the policy name for later use.

    - If not, click **Create Policy**, and define the policy using the following policy template.

        ```json
        {
        "Version": "1",
        "Statement": [
            {
            "Effect": "Allow",
            "Action": [
                "oss:PutObject"
            ],
            "Resource": "acs:oss:*:*:<Your-Bucket-Name>/*"
            }
        ]
        }
        ```

    Replace `<Your-Bucket-Name>` with the name of your OSS bucket where TiDB Cloud will write audit logs. For example, if your bucket name is `auditlog-bucket`, use: `"Resource": "acs:oss:*:*:auditlog-bucket/*"`.

> 3. In the Alibaba Cloud Console, go to **RAM** > **Identities** > **Roles**, and then check whether a role already exists whose **trusted entity** matches the TiDB Cloud Account ID and External ID you recorded earlier.

Reviewer: ```suggestion
3. In the Alibaba Cloud console, go to **RAM** > **Identities** > **Roles**, and then check whether a role already exists whose **trusted entity** matches the TiDB Cloud Account ID and External ID you recorded earlier.
```

    - If yes, record the role name for later use.

    - If not, click **Create Role** by taking the following steps.

        1. In the role creation page, click **Switch to Policy Editor**.
        2. Under **Principal**, choose **Cloud Account** and enter the **TiDB Cloud Account Id** in the field.
        3. Under **Action**, select **sts:AssumeRole** from the drop-down list.
        4. Click **Add condition**, and then configure the condition as follows:
            - Set **Key** to ``sts:ExternalId``.
            - Set **Operator** to ``StringEquals``.
            - Set **Value** to the **TiDB Cloud External ID**.
        5. Click **OK** to open the **Create Role** dialog.
        6. Enter the role name in the **Role Name** field, and lick **OK** to create the role.

4. After the role is created, go to the **Permissions** tab and click **Grant Permission**.

    In the dialog, configure the following settings:

    - For **Resource Scope**, select **Account**.

>     - In the **Policy** field,  select the OSS write policy created earlier.

Reviewer: ```suggestion
    - In the **Policy** field, select the OSS write policy created earlier.
```

    You can add one audit rule at a time. Each rule specifies a user expression, database expression, table expression, and access type. You can add multiple audit rules to meet your auditing requirements.

2. In the **Log Filter Rules** section, click **>** to expand and view the list of audit rules you have added.

> **Note:**
>
> - The filter rules are regular expressions and case-sensitive. If you use the wildcard rule `.*`, all users, databases, or table events in the instance are logged.
> - Because audit logging consumes instance resources, be prudent when specifying filter rules. To minimize the consumption, it is recommended that you specify filter rules to limit the scope of audit logging to specific database objects, users, and actions, where possible.

## View audit logs

By default, TiDB Cloud stores database audit log files in your storage service, so you need to read the audit log information from your storage service.

TiDB Cloud audit logs are readable text files with the instance ID, Pod ID, and log creation date incorporated into the fully qualified filenames.

> For example, `13796619446086334065/tidb-0/tidb-audit-2022-04-21T18-16-29.529.log`. In this example, `13796619446086334065` indicates the instance ID and `tidb-0` indicates the Pod ID.

Reviewer 2: For premium TiDB, the audit log does not name as `tidb-0`, besides don't use `Pod` here.

---
title: TiDB Cloud Premium Database Audit Logging

> summary: Learn about how to audit a instance in TiDB Cloud.

Reviewer 3: ```suggestion
title: "{{{ .premium }}} Database Audit Logging"
summary: Learn how to audit a instance in {{{ .premium }}}.
```

---

> # TiDB Cloud Premium Database Audit Logging

Reviewer 3: ```suggestion
# {{{ .premium }}} Database Audit Logging
```
